wallet/src/main/java/com/android/identity_credential/wallet/ui/SelfieRecorder.kt

```diff
@@ -0,0 +1,157 @@
+package com.android.identity_credential.wallet.ui
+
+import android.content.ContentValues
+import android.os.Build
+import android.provider.MediaStore
+import androidx.activity.ComponentActivity
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.MirrorMode.MIRROR_MODE_ON_FRONT_ONLY
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.video.FallbackStrategy
+import androidx.camera.video.MediaStoreOutputOptions
+import androidx.camera.video.Quality
+import androidx.camera.video.QualitySelector
+import androidx.camera.video.Recorder
+import androidx.camera.video.Recording
+import androidx.camera.video.VideoCapture
+import androidx.camera.video.VideoRecordEvent
+import androidx.core.content.ContextCompat
+import com.android.identity.util.Logger
+import kotlinx.coroutines.guava.await
+import java.text.SimpleDateFormat
+
+
+/**
+ * Video recorder to record a selfie video for identity verification.
+ */
+class SelfieRecorder(
+    private val activity: ComponentActivity,
+    private val onRecordingStarted: () -> Unit,
+    private val onFinished: (ByteArray) -> Unit
+) {
+    companion object {
+        private const val TAG = "SelfieRecorder"
+        private const val FILENAME_TIME_FORMAT = "yyyy-MM-dd-HH-mm-ss-SSS"
+        private const val FILENAME_FORMAT = "VerificationSelfie-\$datetime.mp4"
+    }
+
+    private lateinit var cameraProvider: ProcessCameraProvider
+    private lateinit var videoCapture: VideoCapture<Recorder>
+    private var recording: Recording? = null
+
+    /**
+     * Starts camera and prepares to record and return a video.
+     */
+    suspend fun launchCamera(surfaceProvider: Preview.SurfaceProvider) {
+        if (::cameraProvider.isInitialized) {
+            throw IllegalStateException("Camera already started.")
+        }
+        cameraProvider = ProcessCameraProvider.getInstance(activity).await()
+
+        // Configure preview, so the user can see what they're recording:
```

Comment:
Did you mean to punctuate all of these comments with a colon?

Reply:
I did, since they're introducing the next block of code... But if that's distracting, I can change them to a period...?